projects/plugins/boost/app/modules/optimizations/page-cache/pre-wordpress/storage/class-file-storage.php

```diff
@@ -152,33 +163,39 @@ private function sanitize_path( $path ) {
 	}
 
 	/**
-	 * Delete all cached data for the given path.
+	 * Delete cache based on given parameters.
 	 *
-	 * @param string $path - The path to delete. File or directory.
-	 * @param string $type - defines what files/directories are deleted or rebuilt.
-	 * @return bool|Boost_Cache_Error True on success, error object on failure
+	 * @param string $path - The path to delete the cache for.
+	 * @param array  $args - The parameters defining the cache filename.
+	 * Example:
+	 * array(
+	 *  'rebuild' => (boolean) default true - If true, cache files will be rebuilt instead of being deleted.
+	 *  'parameters' => false | array, default false - If array is provided, the files created for that specific request matching the parameter will be effected. If parameter is provided, recursive is ignored and considered as false.
+	 *  'recursive' => (boolean) default false - If true, the cache will be deleted recursively in subdirectories.
+	 * )
 	 */
-	public function invalidate( $path, $type ) {
-		Logger::debug( "invalidate: $path $type" );
-		$normalized_path = $this->root_path . Boost_Cache_Utils::normalize_request_uri( $path );
-
-		$response = null;
-		if ( ! in_array( $type, array( Filesystem_Utils::DELETE_FILE, Filesystem_Utils::REBUILD_FILE ), true ) && is_dir( $normalized_path ) ) {
-			$response = Filesystem_Utils::walk_directory( $normalized_path, $type );
-		} elseif ( $type === Filesystem_Utils::DELETE_FILE && is_file( $normalized_path ) ) {
-			$response = Filesystem_Utils::delete_file( $normalized_path );
-		} elseif ( $type === Filesystem_Utils::REBUILD_FILE && is_file( $normalized_path ) ) {
-			$response = Filesystem_Utils::rebuild_file( $normalized_path );
-		}
+	public function clear( $path, $args = array() ) {
+		$normalized_path = Boost_Cache_Utils::trailingslashit( $this->root_path . Boost_Cache_Utils::normalize_request_uri( $path ) );
+
+		$recursive  = $args['recursive'] ?? false;
+		$rebuild    = $args['rebuild'] ?? true;
+		$parameters = $args['parameters'] ?? false;
 
-		if ( $response === null ) {
-			return new Boost_Cache_Error( 'no-cache-files-to-delete', 'No cache files to delete.' );
+		$action = new Simple_Delete();
```

Comment:
Nitpick: I know it's a little thing, but if we put this in the `else` we wouldn't always have to create an instance and then sometimes override it.

Reply:
I was thinking of doing that, but forgot to change that. Thanks for the reminder. :)